http-clients/apache-client/src/main/java/software/amazon/awssdk/http/apache/internal/SdkProxyRoutePlanner.java

```diff
@@ -40,7 +41,7 @@ public class SdkProxyRoutePlanner extends DefaultRoutePlanner {
     public SdkProxyRoutePlanner(String proxyHost, int proxyPort, String proxyProtocol, Set<String> nonProxyHosts) {
         super(DefaultSchemePortResolver.INSTANCE);
         proxy = new HttpHost(proxyHost, proxyPort, proxyProtocol);
-        this.hostPatterns = nonProxyHosts;
+        this.hostPatterns = Collections.unmodifiableSet(nonProxyHosts);
```

Comment:
Removing for now since I will raise another PR just for EI_EXPOSE_REP.
However I think this is needed even for internal classes since we dont want to accidentally update the list when used interballys